src/iuse_actor.cpp

```diff
@@ -1078,16 +1078,28 @@ void reveal_map_actor::load( JsonObject &obj )
     radius = obj.get_int( "radius" );
     message = obj.get_string( "message" );
     JsonArray jarr = obj.get_array( "terrain" );
+    std::string ter;
+    ot_match_type ter_match_type;
     while( jarr.has_more() ) {
-        omt_types.push_back( jarr.next_string() );
+        if( jarr.test_string() ) {
+            ter = jarr.next_string();
+            ter_match_type = ot_match_type::contains;
+        } else {
+            JsonObject jo = jarr.next_object();
+            ter = jo.get_string( "om_terrain" );
+            ter_match_type = jo.get_enum_value<ot_match_type>( jo.get_string( "om_terrain_match_type",
+                             "CONTAINS" ), ot_match_type::contains );
```

Comment:
Seems like `get_enum_value` here isn't doing what we want for it to do. It checks for a **member** called `TYPE`, `EXACT` and any other enum from `ot_match_type` enum list. Obviously it didn't find anything like that, so it falls back to default value, `CONTAINS`.

Reply:
Right, it should not use `get_string`, just do `get_enum_value<>` directly.